core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -424,6 +438,10 @@ private void installConfig(PluginInfo info, Path tmpConfigDir, Path destConfigDi
         // create the plugin's config dir "if necessary"
         Files.createDirectories(destConfigDir);
 
+        final PosixFileAttributes destConfigDirAttributes =
+                Files.getFileAttributeView(destConfigDir.getParent(), PosixFileAttributeView.class).readAttributes();
+        setOwnerGroup(destConfigDir, destConfigDirAttributes);
```

Comment:
Owner and group is not controlled by the umask. The umask continues to dictate the actual privileges and note that I modified the tests to actually check that the permissions are that which would be inherited from the umask.

However, this might be unnecessary *after* the changes that I made to the ownership of the `plugins` folder. I'll revisit and verify.